enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,179 @@
+---
+title: sdk-metrics
+  - "@varshaprasad96"
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+approvers:
+  - "@dmesser" 
+  - "@joelanford"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implementable
+---
+
+# Operator SDK metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The enhancement is intended to define the bundle resources in which SDK labels can be added, so that they can be read by OLM and included in the current OLM metrics. These metrics will help us in providing a measure to evaluate the reachability of Operator SDK among community operators.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has a pipeline built to obtain metrics on operators and report it to the RedHat telemetry, we intend to utilize the same to obtain data on operators using SDK to build bundle images.
+
+### Goals
+
+The goal of this enhancement is to :
+1. List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+The proposed SDK stamps will contain data related to:
+* Version of SDK used
+* SDK project layout
+
+The stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include stamps in the form of `LABEL` which will follow the following format:
+
+```
+LABEL operators.operatorframework.io.metrics.mediatype.v1 = “metrics+v1”
+LABEL operators.operatorframework.io.metrics.builder = “operator-sdk-v0.17.0”
+LABEL operators.operatorframework.io.metrics.project_layout = “sdk_project_layout/layout_version”
+```
+
+2. annotations.yaml
+
+Similar to the format followed in dockerfile, `annotations.yaml` will have the following labels:
+
+```
+operators.operatorframework.io.metrics.mediatype.v1 = “metrics+v1”
+operators.operatorframework.io.metrics.builder = “operator-sdk-v0.17.0”
```

Comment:
These examples have the `metrics` prefix, and the labels further down do not. Is the proposal to have different keys for image labels vs. CSV labels?

Reply:
I think the intent is to keep the annotation key as short as possible so as to not approach the 63 character limit, and having `metrics` in the key doesn't add much information to annotation keys specifically since they should be treated opaquely by users.